homework-4/classes/CollectionDictionary.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CollectionApplication
+{
+    class CollectionDictionary
+    {
+        private Dictionary<string, int> elements = new Dictionary<string, int>();
+
+        public void Menu()
+        {
+            while (true)
+            {
+                Console.WriteLine("\nPlease enter a number from the given menu:");
+                Console.WriteLine("1. Add an element to the Dictionary.");
+                Console.WriteLine("2. Print all the elements present in the Dictionary.");
+                Console.WriteLine("3. Delete the last element from the Dictionary.");
+                Console.WriteLine("4. Delete the first element from the Dictionary.");
+                Console.WriteLine("5. Delete the middle element from the Dictionary.");
+                Console.WriteLine("6. Calculate the average of the elements present in the Dictionary.");
+                Console.WriteLine("7. Go back to the previous menu.");
+                Console.WriteLine("8. Exit the application.");
+
+                if (!int.TryParse(Console.ReadLine(), out int choice))
+                {
+                    Console.WriteLine("Please enter a valid number.");
+                    continue;
+                }
+
+                switch (choice)
+                {
+                    case 1:
+                        AddElement();
+                        break;
+                    case 2:
+                        PrintElements();
+                        break;
+                    case 3:
+                        DeleteLastElement();
+                        break;
+                    case 4:
+                        DeleteFirstElement();
+                        break;
+                    case 5:
+                        DeleteMiddleElement();
+                        break;
+                    case 6:
+                        CalculateAverage();
+                        break;
+                    case 7:
+                        return;
+                    case 8:
+                        Console.WriteLine("Exiting the application...");
+                        Environment.Exit(0);
+                        break;
+                    default:
+                        Console.WriteLine("Please enter a valid number.");
+                        break;
+                }
+            }
+        }
+
+        private void AddElement()
+        {
+            Console.WriteLine("Enter the key:");
+            string key = Console.ReadLine();
+            Console.WriteLine("Enter the value:");
+            if (int.TryParse(Console.ReadLine(), out int value))
+            {
+                elements[key] = value;
```

Comment:
what if the key is already present in the dictionary? This will overwrite that.. I know its not mentioned in the requirement but you should think of these small edge cases and implementation depends upon you. If you want to overwrite that value just have a comment here that if the key is already present it will be overwritten... or you can tell the user that key is already present.